crypto/hpke.c

```diff
@@ -0,0 +1,2730 @@
+    {15, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    { HPKE_KEM_ID_P256,
+      "EC", "P-256",
+      NID_X9_62_prime256v1, EVP_sha256,
+      32, 65, 65, 32 }, /* maybe "prime256v1" instead of P-256? */
+    { HPKE_KEM_ID_P384,
+      "EC", "P-384",
+      NID_secp384r1, EVP_sha384,
+      48, 97, 97, 48 },
+    { HPKE_KEM_ID_P521,
+      "EC", "P-521",
+      NID_secp521r1, EVP_sha512,
+      64, 133, 133, 66 },
+    {19, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {20, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {21, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {22, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {23, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {24, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {25, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {26, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {27, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {28, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {29, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {30, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    {31, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+    { HPKE_KEM_ID_25519,
+      "X25519", NULL,
+      EVP_PKEY_X25519, EVP_sha256,
+      32, 32, 32, 32 },
+    { HPKE_KEM_ID_448,
+      "X448", NULL,
+      EVP_PKEY_X448, EVP_sha512,
+      64, 56, 56, 56 },
+    {34, NULL, NULL, 0, NULL, 0, 0, 0 }, /* keep indexing correct */
+};
+
+
+#if defined(SUPERVERBOSE) || defined(TESTVECTORS)
+/*
+ * @brief table of KEM strings
+ *
+ * Note: This also need blanks
+ */
+const char *hpke_kem_strtab[] = {
+    NULL, NULL, NULL, NULL,
+    NULL, NULL, NULL, NULL,
+    NULL, NULL, NULL, NULL,
+    NULL, NULL, NULL, NULL,
+    HPKE_KEMSTR_P256,
+    HPKE_KEMSTR_P384,
+    HPKE_KEMSTR_P521,
+    NULL,
+    NULL, NULL, NULL, NULL,
+    NULL, NULL, NULL, NULL,
+    NULL, NULL, NULL, NULL,
+    HPKE_KEMSTR_X25519,
+    HPKE_KEMSTR_X448,
+    NULL};
+#endif
+
+/*!
+ * @brief info about a KDF
+ */
+typedef struct {
+    uint16_t       kdf_id; /**< code point for KDF */
+    const EVP_MD*  (*hash_init_func)(void); /**< the hash alg we're using */
+    size_t         Nh; /**< length of hash/extract output */
+} hpke_kdf_info_t;
+
+/*!
+ * @brief table of KDFs
+ */
+static hpke_kdf_info_t hpke_kdf_tab[] = {
+    { 0, NULL, 0 }, /* keep indexing correct */
+    { HPKE_KDF_ID_HKDF_SHA256, EVP_sha256, 32 },
+    { HPKE_KDF_ID_HKDF_SHA384, EVP_sha384, 48 },
+    { HPKE_KDF_ID_HKDF_SHA512, EVP_sha512, 64 }
+};
+
+#if defined(SUPERVERBOSE) || defined(TESTVECTORS)
+/*
+ * @brief table of KDF strings
+ */
+const char *hpke_kdf_strtab[] = {
+    NULL,
+    HPKE_KDFSTR_256,
+    HPKE_KDFSTR_384,
+    HPKE_KDFSTR_512};
+#endif
+
+static OSSL_LIB_CTX *hpke_libctx = NULL;
```

Comment:
!

Reply:
that one's fixed in https://github.com/openssl/openssl/pull/17172/commits/ededd5fefacd063082fd6eefcc0b94727eeb0f3e - the OSS_LIB_CTX pointer is now passed down as needed to the internal funcs